update deps

syuilo opened the pull request:
<!-- ℹ お読みください / README
PRありがとうございます！ PRを作成する前に、コントリビューションガイドをご確認ください:
Thank you for your PR! Before creating a PR, please check the contribution guide:
https://github.com/misskey-dev/misskey/blob/develop/CONTRIBUTING.md
-->

## What
<!-- このPRで何をしたのか？ どう変わるのか？ -->
<!-- What did you do with this PR? How will it change things? -->

## Why
<!-- なぜそうするのか？ どういう意図なのか？ 何が困っているのか？ -->
<!-- Why do you do it? What are your intentions? What is the problem? -->

## Additional info (optional)
<!-- テスト観点など -->
<!-- Test perspective, etc -->

## Checklist
- [ ] Read the [contribution guide](https://github.com/misskey-dev/misskey/blob/develop/CONTRIBUTING.md)
- [ ] Test working in a local environment
- [ ] (If needed) Add story of storybook
- [ ] (If needed) Update CHANGELOG.md
- [ ] (If possible) Add tests

syuilo committed:
wip

syuilo commented:
https://vite.dev/guide/migration.html#migration-from-v5

kakkokari-gtyih committed:
Merge branch 'develop' into update-deps-20250120

kakkokari-gtyih committed:
bump misskey-dev/eslint-plugin

kakkokari-gtyih committed:
lint fixes (backend)

kakkokari-gtyih committed:
lint fixes (frontend)

kakkokari-gtyih committed:
lint fixes (frontend-embed)

kakkokari-gtyih committed:
rollback nsfwjs to 4.2.0

ref: infinitered/nsfwjs#904

kakkokari-gtyih committed:
rollback openapi-typescript to v6

v7でOpenAPIのバリデーションが入るようになった関係でスコープ外での変更が避けられないため一時的に戻した

kakkokari-gtyih committed:
lint fixes (misskey-js)

kakkokari-gtyih committed:
temporarily disable errored lint rule (frontend-shared)

kakkokari-gtyih committed:
fix lint

anatawa12 commented:
nsfwjs は #15248 で言及し忘れたけど https://github.com/infinitered/nsfwjs/pull/910 を踏んだので意図的に落としてた。どっかにバージョンを latest にしない理由をまとめる仕組みあったほうがいいかもな...

kakkokari-gtyih committed:
temporarily ignore errored file for lint (frontend-shared)

kakkokari-gtyih committed:
rollback simplewebauthn/server to 12.0.0

v13 contains breaking changes that require some decision making

kakkokari-gtyih committed:
lint fixes (frontend-shared)

kakkokari-gtyih committed:
build misskey-js with types

kakkokari-gtyih committed:
fix(backend): migrate simplewebauthn/server to v12

kakkokari-gtyih committed:
fix(misskey-js/autogen): ignore indent rules to generate consistent output

kakkokari-gtyih committed:
attempt to fix test

changes due to capricorn86/happy-dom#1617 (XMLSerializer now produces valid XML)

kakkokari-gtyih committed:
attempt to fix test

changes due to capricorn86/happy-dom#1617 (XMLSerializer now produces valid XML)

kakkokari-gtyih committed:
fix test

kakkokari-gtyih committed:
fix test

kakkokari-gtyih committed:
fix test

anatawa12 commented on packages/frontend/src/components/MkCaptcha.vue:
```diff
@@ -50,6 +50,7 @@ type CaptchaContainer = {
 };
 
 declare global {
+	// eslint-disable-next-line @typescript-eslint/no-empty-object-type
```
```suggestion
	// Window を拡張してるため、空ではない
	// eslint-disable-next-line @typescript-eslint/no-empty-object-type
```

anatawa12 commented on packages/frontend-shared/eslint.config.js:
```diff
@@ -97,4 +97,11 @@ export default [
+	{
+		ignores: [
+			// TODO: Error while loading rule '@typescript-eslint/naming-convention': Cannot use 'in' operator to search for 'type' in undefined のため一時的に無効化
+			// See https://github.com/misskey-dev/misskey/pull/15311
+			'js/i18n.ts',
```
Upstreamに報告して / 見つけてそっちへのリンクを張りたいとおもったけど eslint の更新してないのが原因だと考えられるから一旦これでも良さそう。
eslint v9 / typescript-eslint v8のissueもないから切ったほうが良さそうですね。

kakkokari-gtyih committed:
Apply suggestions from code review

Co-authored-by: anatawa12 <anatawa12@icloud.com>

samunohito commented:
> 動くかどうかは分からない

下記理由から、 2025.1.0 ではなく次以降のバージョンに取り込むのが良いと考えますが、どうでしょうか
- WebAuthnService.ts や MfmService.ts に対して変更が入っている
- 各依存パッケージの更新による副作用を検証する期間が短い
- 2025.1.0 には重要なbugfixが多くあり、出来る限り他のバグを埋め込むリスクを下げたい

samunohito commented:
それなら、該当パッケージをリストアップして最低限にとどめた方がリスクが低いように思えますが…
まあ、お任せします

kakkokari-gtyih committed:
Merge remote-tracking branch 'msky/develop' into update-deps-20250120

syuilo commented:
このバージョンのnodemonに不具合あるかも(プロセスが終了しない)

syuilo commented:
ブランチ切り替えた時にpnpm iし忘れていた可能性がある(ほかのブランチでpnpm iしたら再現しなくなった)

kakkokari-gtyih committed:
Merge branch 'develop' into update-deps-20250120

kakkokari-gtyih committed:
Merge branch 'develop' into update-deps-20250120

kakkokari-gtyih committed:
bump summaly to v5.2.0

syuilo commented:
二回連続でpnpm devすると発生することが判った(nodemonのバージョンは無関係だった)

syuilo commented:
かも

kakkokari-gtyih committed:
Merge branch 'develop' into update-deps-20250120

kakkokari-gtyih committed:
update tabler-icons to v3.30.0-based
